pkg/domain/attributor.go

```diff
@@ -37,3 +38,49 @@ type NexposeAttributedAssetVulnerabilities struct {
 	Vulnerabilities []AssetVulnerabilityDetails `json:"assetVulnerabilityDetails"`
 	BusinessContext CloudAssetDetails           `json:"businessContext"`
 }
+
+// AssetNotFoundError occurs when a request to an asset inventory system
+// returns either a 404 Not Found response, or a 200 OK response with no results
+type AssetNotFoundError struct {
+	Inner          error
+	AssetID        string
+	ScanTimestamp  string
+	AssetInventory string
+}
+
+func (err AssetNotFoundError) Error() string {
+	return fmt.Sprintf(
+		"Result not found for asset with ID %s as of scan time %s in asset inventory %s: %v",
+		err.AssetID, err.ScanTimestamp, err.AssetInventory, err.Inner)
+}
+
+// AssetInventoryRequestError occurs when a request to an asset inventory system
+// returns a failure response
+type AssetInventoryRequestError struct {
+	Inner          error
+	AssetID        string
+	ScanTimestamp  string
+	AssetInventory string
+	Code           int
+}
+
+func (err AssetInventoryRequestError) Error() string {
+	return fmt.Sprintf(
+		"Request to asset inventory %s failed with code %d for asset with ID %s as of scan time %s: %v",
+		err.AssetInventory, err.Code, err.AssetID, err.ScanTimestamp, err.Inner)
+}
+
+// AssetInventoryMultipleAssetsFoundError occurs when a request to an asset inventory system
+// returns a successful response with multiple assets
+type AssetInventoryMultipleAssetsFoundError struct {
```

Comment:
Question: 
What does this mean in a business context that the asset inventory system returned multiple assets?

Reply:
I don't know that I can answer your question broadly, but the reason it's being treated as an error is because if an asset inventory returns multiple assets, we have no mechanism, at the moment, to determine which asset is the "correct" one (i.e., the asset we actually scanned).

More generally, there are multiple reasons any asset inventory might return multiple results - we're querying by IP and hostname, which are not necessarily globally unique values, ordering issues with asset change events feeding the inventory system, etc.

Does that help, or do you need more context?

Reply:
I think so. What does that mean for the plan moving forward then? If we start seeing lots of cases where multiple assets are returned, will we prioritize this as something to figure out and work on? It feels bad dropping these

Reply:
I'll defer to @faldridge, but that's the intention behind throwing the error here - that this is something we should monitor and alert on, so we can determine how common this case actually is, and how we would resolve this given actual data.

Reply:
Yep, @bkrebsbach is spot on here. Similarly, we are collecting data on cases where attribution succeeds, but issue creation fails, because the service owner is not a user in the ticketing system. Example: "billing-admin-team@tech-org.com". 

We are starting to see this in the temporary system we're using while we complete this one.